README.md

```diff
@@ -1,41 +1,79 @@
 
 ### To Do
 * Literally everything
-* Fix up URL management to resolve references etc
-* Env var configuration (config uniformity too)
-* The rest of the Origin service
+* List is ever changing
+
+# Testing
+The basic tests are surfaced via the Makefile
+
+* `make test`
+  * Runs Unit tests and ignores Integration suite
+  * Requires no Environment variables
+  * Ex. `make test`
+* `make cover`
+  * Load go cover details in your browser
+* `make integration`
+  * Runs integration suite against Striketracker API
+  * Requires `INTEGRATIONACCOUNTHASH` & `AUTHORIZATIONHEADERKEY` environment variables
+    * INTEGRATIONACCOUNTHASH
+      * The account hash of the subaccount you are running integration tests against
+    * AUTHORIZATIONHEADERKEY
+      * The authorization header key for authenticated API access
+  * Ex. `INTEGRATIONACCOUNTHASH=f98fsj32k AUTHORIZATIONHEADERKEY=fj32jk43kj32kj3rkhj make integration`
 
 # Usage
 You will need your authorizationHeaderToken from Highwinds as well as manage your own accountHashes.
 
-There is no plan to implement username/password based authentication. While it is supported by the Striketracker API, this library is focused on using a permaent API token.
+There is no plan to implement username/password based authentication. While it is supported by the Striketracker API, this library is focused on using a permanent API token.
 
 ### Client
-You must first configure and maintain a client in your application. As of right now this looks like the example below, however it's going to accept EnvVars ultimately for secrets (not implemented yet)
+You must first configure and maintain a client in your application. There are a few ways to do this.
 
 `import "github.com/openwurl/wurlwind/striketracker"`
 
+Via Functional Parameters
 ```
-c := striketracker.NewClient(debug (bool), authorizationHeaderToken (string), application-name (string))
+c, err := striketracker.NewClientWithOptions(
+    striketracker.WithApplicationID("DescriptiveApplicationName"),
+    striketracker.WithDebug(true),
+    striketracker.WithAuthorizationHeaderToken(stringAuthToken),
+    striketracker.WithRequestTimeout(intSeconds),
+)
 ```
 
-ToDo:
-
-* `NewClientFromConfig`
-  * config-struct functional driven configuration
-  * (ex. `NewClientFromConfig(WithAuthorizationToken(token)))`)
-
+Via user-defined configuration
+```
+c, err := striketracker.NewClientFromConfiguration(&striketracker.Configuration{
+	Debug: false,
+	AuthorizationHeaderToken: yourtoken,
+	Timeout: TimeOutInSeconds,
+	ApplicationID: YourApplicationName,
+})
+```
 
 # Services
 A brief overview of the Highwinds services exposed in this API Client Library
 
+### Context
+Context can be passed into most, if not all service methods - to be used for early cancellation or to configure a timeout per operation.
+
+```
+ctx := context.Background()
+ctx, cancel := context.WithTimeout(ctx, 3*time.Second)
+defer cancel()
+
+resp, err := service.Action(ctx, ...)
+```
+
```

Comment:
I don't quite understand the term Context, it appears by my understanding of the docs to be only providing a timeout capability.  Is Context going to have more functionality in the future?

Also, in my experience with 'Context' objects/parameters, it's always seemed like a hack to 'cleanly' pass info down the call chain of methods.  It works...however, in my experience, it just causes a leaky abstraction (https://en.wikipedia.org/wiki/Leaky_abstraction) and can get abused...so just be judicious with what it's used for.

Reply:
context.Context is a golang best practice.

https://golang.org/pkg/context/

https://medium.com/@cep21/how-to-correctly-use-context-context-in-go-1-7-8f2c0fafdf39

There is no value gain for us to use it for data, it's essentially a cancellation channel

Reply:
```
ctx := context.Background()
ctx, cancel := context.WithTimeout(ctx, 3*time.Second)
defer cancel()
```

This, for instance, if injected, supplies a 3 second context timeout, or if cancel() was called before it completed (in this case cancel() is deferred to fire at the end of the method regardless) - cancel() could still be called if the consumer placed this call into a goroutine and used channels for the response (the pattern one would use in say a webserver or service it was implemented in)

Reply:
While context is generally used to inform lower levels (never block), and is even found frequently in libraries such as gorilla mux and even the standard library now that it is the go-to for concurrency/threading - we are using it just for cancellation at the moment - however that may change depending on the complexity of interactions with the API. up in the air right now

Reply:
it's nice that your using an established lib.  I've hand rolled this in the past and been bitten by it :)

Reply:
It's actually standard library ;) ;) ;)

It's why it's under pkg/